plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/DataflowMavenModel.java

```diff
@@ -193,29 +194,9 @@ public void trackDataflowDependency(IProgressMonitor monitor) throws CoreExcepti
      * provides that namespace for the prefix 'pom'.
      */
     @VisibleForTesting
-    static final NamespaceContext POM_NS_CONTEXT = new NamespaceContext() {
-      @Override
-      public String getNamespaceURI(String prefix) {
-        if (prefix == null) {
-          throw new NullPointerException("Null prefix");
-        } else if ("pom".equals(prefix)) {
-          return "http://maven.apache.org/POM/4.0.0";
-        } else if ("xml".equals(prefix)) {
-          return XMLConstants.XML_NS_URI;
-        }
-        return XMLConstants.NULL_NS_URI;
-      }
-
-      @Override
-      public String getPrefix(String namespaceURI) {
-        throw new UnsupportedOperationException();
-      }
-
-      @Override
-      public Iterator<?> getPrefixes(String namespaceURI) {
-        throw new UnsupportedOperationException();
-      }
-    };
+    static final NamespaceContext POM_NS_CONTEXT = new MappedNamespaceContext(ImmutableMap.of(
```

Comment:
Why is this visible for testing?

Reply:
`DataflowMavenModelTest` has been making use of it. I also added tests to check the `NamespaceContext` itself, as it was already `@VisibleForTesting`.